### ISSUE TITLE
Local Network Image Previews Are Broken

### ISSUE THREAD

**Issue by joeyv120** [Zettlr/Zettlr#5495]
### Description

Installed Zettlr for the first time (version 3.3.1) and image previews were not loading. I noticed that they were working in the "Zettlr Tutorial" workspace that comes preinstalled. After learning that a similar bug ([#5489](https://github.com/Zettlr/Zettlr/issues/5489)) in 3.3.0 was fixed in 3.3.1 I installed 3.2.(?) and found that all my image previews were loading as expected. I installed 3.3.0, then auto-updated to 3.3.1, and in both versions the previews were broken.

It seems that this issue is specific to **previews for images located on a local server**. Clicking "open image externally" opens the remote image in an external program as expected, and links to other remote file types open as expected. However, there are no previews for images located on a local server. Image previews work for images located on a web server.

The location of the md file seems to be irrelevant.

### Reproducing

Add a local server-hosted image reference to a .md file; the preview will **NOT** load, but clicking it will open the file. For example:

`![Some Server Image](\\remoteHost\fileName.jpg)`

Add a local image reference to a .md file; the preview **will** load. For example:

`![Some Local Image](C:\fileName.jpg)`

Add a web server image reference to a .md file; the preview **will** load. For example:

`![Some web server image](https://en.wikipedia.org/static/images/icons/wikipedia.jpg)`

UPDATE (as I'm writing this): the following syntax **will** load the preview for a local server-hosted image, but will **NOT** open the file when clicking on it:

`![re-formatted syntax](file:///remoteHost/fileName.jpg)`

If the MD file and the image file are in the same remote folder, this issue persists. For example, the following preview will **NOT** load, but clicking the image will open the file externally as expected:

`![MD file and image file in the same remote folder](fileName.jpg)`

### Zettlr version

3.3.1

### Installation Method

From the Website or GitHub

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit (includes Apple Silicon)

### Operating System Version

Windows 11

**Comment by nathanlesage** [comment]
That should be a simple fix after my findings from fixing the one addressed by 3.3.1.

**Comment by nathanlesage** [comment]
@kyaso Do you maybe have a network drive at home and the chance test this? I'm 99% sure the paths are just malformed because I forgot an edge case. It's probably the automagic slicing and adding of slashes in front of path names in the file protocol handler for Windows…

**Comment by kyaso** [comment]
> Do you maybe have a network drive at home and the chance test this?

No, unfortunately I don't have a network drive :(

**Comment by nathanlesage** [comment]
Alright, in that case I'll see to it, maybe I can cannibalize one of my USB keys to get a network share via my FritzBox on the weekend

**Comment by nathanlesage** [comment]
I've been on it this morning – no solution yet, but I really want to share this tidbit from the RFC with you:

> "A relative reference that begins with two slash carachters is termed a network-path reference; **such references are rarely used**." ([RFC 3986, § 4.2](https://datatracker.ietf.org/doc/html/rfc3986#section-4.2))

**Comment by nathanlesage** [comment]
Additional observations:

Writing `![re-formatted syntax](file:///remoteHost/fileName.jpg)` will result in the main process attempting to load the file `/remoteHost/fileName.jpg`. This means that, on Windows, file shares are apparently mounted similarly to UNIX volumes, and instead of `C:/` you have `/remoteHost`.

However, just keeping it like that would break on macOS and Linux, since there, network shares are usually mounted at, e.g., `/mnt/remoteHost/` and `/Volumes/remoteHost` respectively.

I was finally able to test it out at home (thanks AVM for including a basic Samba share in your Fritz!Box), and valid paths to the default image on its internal drive are:

* `smb://fritz.box/FRITZ.NAS/Bilder/FRITZ-Picture.jpg`
* `/Volumes/FRITZ.NAS/Bilder/FRITZ-Picture.jpg`, and thus also:
* `file:///Volumes/FRITZ.NAS/Bilder/FRITZ-Picture.jpg`

Cool, okay. Luckily I also didn't forget my work laptop, so on Linux (at least Manjaro w/ Dolphin) the very same picture on the very same device will have the following valid links:

* `smb://fritz-user@box-ip-address/FRITZ.NAS/Bilder/Fritz-Picture.jpg`, where `fritz-user` is the user account of the Fritz!Box and the `box-ip-address` is the corresponding local IP address
* The address won't be automatically mounted anywhere, however, and even then, with `cifs` you can literally mount that bastard wherever you want.
* Typing `smb://` and the other yadda yadda will have Firefox google for the smb link instead of trying to somehow open that 🥴

I wish I could say just adding `smb:` in front of the network share would work. But that would break on Windows and very apparently on Linux, since `smb` is the Samba-protocol that is literally defined to only exist on non-Windows machines to be able to talk to Windows network shares. 🙂🔫 Also, this helps us NOT AT ALL when it comes to actually loading the file, since the main process MUST perform some lookup, meaning that we end up with the following:

* No lookup of network shares using the `//remoteHost`-syntax on Linux, one has to manually mount and then provide the local path
* `smb://remoteHost/fileName.jpg` (with or without the `smb:` in front) should resolve to `/Volumes/remoteHost/fileName.jpg` on macOS machines
* `file:///remoteHost/fileName.jpg` should resolve to `/remoteHost/fileName.jpg` on Windows

In other words, being on Windows, Zettlr should search for the same path, but with only a single slash in front, on macOS it should just prepend /Volumes, and on Linux it should do nothing because there's nothing to find.

Let me see if I can somehow mould the corresponding functions in some way to do that ...

**Comment by nathanlesage** [comment]
It feels really hacky, but it works for displaying an image from a network share.

![image](https://github.com/user-attachments/assets/ebcb41cc-09e4-4a51-afe9-4638b1d25710)

On Windows, this should now even open the file appropriately, but not on macOS. But I'm gonna go with Tim Berners-Lee: "It's used rarely."

**Comment by OTHGabriel** [comment]
I seem to be having the same problem with relative file paths,

`./fig/image.png`

but none of the syntax prefixes from above could resolve that issue, and it would probably also break the file should I ever move the folder to a local hard drive.
